### DISCUSSION TITLE
Prevent switching foreground mode on App focus/blur

### DISCUSSION THREAD

**Discussion by veitbjarsch** [HaylLtd/cordova-plugin-background-geolocation#140] [Q&A]
Currently if I start the service with `startForeground: false` the plugin will switch automatically to `startForeground: true` once I minimize the app and switches back to `startForeground: false` if I maximize the app again. How can I disable this behavior?

**Comment by HarelM** [comment]
Can you share some logs? Sounds like a bug in general, although this might be intentional, IDK, there's a lot of history to this codebase...

**Comment by veitbjarsch** [comment]
I'll try to share some logs during the next days.

**Comment by AbdullahSohail-SE** [comment]
@HarelM the logs,
[
    {
        "id": 443,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [3]. startId: [4]",
        "timestamp": 1678607567553,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 442,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [8]. startId: [3]",
        "timestamp": 1678607567545,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 441,
        "context": 0,
        "level": "INFO",
        "message": "App will be resumed multitasking=true",
        "timestamp": 1678607567501,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 440,
        "context": 0,
        "level": "INFO",
        "message": "App is visible",
        "timestamp": 1678607567492,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 439,
        "context": 0,
        "level": "INFO",
        "message": "App is no longer visible",
        "timestamp": 1678607562637,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 438,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [4]. startId: [2]",
        "timestamp": 1678607561944,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 437,
        "context": 0,
        "level": "INFO",
        "message": "App will be paused multitasking=true",
        "timestamp": 1678607561906,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 436,
        "context": 0,
        "level": "DEBUG",
        "message": "Received MSG_ON_SERVICE_STARTED",
        "timestamp": 1678607545279,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 435,
        "context": 0,
        "level": "INFO",
        "message": "Requesting location updates from provider gps",
        "timestamp": 1678607545256,
        "logger": "com.marianhello.bgloc.provider.RawLocationProvider"
    },
    {
        "id": 434,
        "context": 0,
        "level": "INFO",
        "message": "Creating RawLocationProvider",
        "timestamp": 1678607545249,
        "logger": "com.marianhello.bgloc.provider.RawLocationProvider"
    },
    {
        "id": 433,
        "context": 0,
        "level": "DEBUG",
        "message": "Will start service with: Config[distanceFilter=1 stationaryRadius=0.0 desiredAccuracy=0 interval=1000 fastestInterval=1000 activitiesInterval=1000 isDebugging=true stopOnTerminate=true stopOnStillActivity=true startOnBoot=false startForeground=false notificationsEnabled=true locationProvider=2 nTitle=Recording nText=Tracking is active nIconLarge= nIconSmall=app_icon_small nIconColor=#30679a url= syncUrl= syncThreshold=100 httpHeaders={} maxLocations=10000 postTemplate=null]",
        "timestamp": 1678607545240,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 432,
        "context": 0,
        "level": "WARN",
        "message": "Attempt to start unconfigured service. Will use stored or default.",
        "timestamp": 1678607545234,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 431,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [NOT STARTED] state. cmdId: [0]. startId: [1]",
        "timestamp": 1678607545221,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 430,
        "context": 0,
        "level": "INFO",
        "message": "Network condition changed has connectivity: true",
        "timestamp": 1678607545219,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 429,
        "context": 0,
        "level": "INFO",
        "message": "Creating PostLocationTask",
        "timestamp": 1678607545197,
        "logger": "com.marianhello.bgloc.PostLocationTask"
    },
    {
        "id": 428,
        "context": 0,
        "level": "INFO",
        "message": "Creating LocationServiceImpl",
        "timestamp": 1678607545173,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 427,
        "context": 0,
        "level": "INFO",
        "message": "Attempt to start bg service",
        "timestamp": 1678607544823,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 426,
        "context": 0,
        "level": "INFO",
        "message": "User granted requested permissions",
        "timestamp": 1678607544812,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 425,
        "context": 0,
        "level": "DEBUG",
        "message": "Starting service",
        "timestamp": 1678607544804,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 424,
        "context": 0,
        "level": "INFO",
        "message": "Getting current location with timeout:2147483647 maximumAge:9223372036854775807 enableHighAccuracy:false",
        "timestamp": 1678607544723,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 423,
        "context": 0,
        "level": "INFO",
        "message": "Deleting all locations",
        "timestamp": 1678607541127,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 422,
        "context": 0,
        "level": "DEBUG",
        "message": "Service configured with: Config[distanceFilter=1 stationaryRadius=0.0 desiredAccuracy=0 interval=1000 fastestInterval=1000 activitiesInterval=1000 isDebugging=true stopOnTerminate=true stopOnStillActivity=true startOnBoot=false startForeground=false notificationsEnabled=true locationProvider=2 nTitle=Recording nText=Tracking is active nIconLarge= nIconSmall=app_icon_small nIconColor=#30679a url= syncUrl= syncThreshold=100 httpHeaders={} maxLocations=10000 postTemplate=null]",
        "timestamp": 1678607540984,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 421,
        "context": 0,
        "level": "DEBUG",
        "message": "Registering event listeners",
        "timestamp": 1678607540685,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 420,
        "context": 0,
        "level": "INFO",
        "message": "Initializing plugin",
        "timestamp": 1678607540035,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 419,
        "context": 0,
        "level": "DEBUG",
        "message": "Destroying RawLocationProvider",
        "timestamp": 1678606203257,
        "logger": "com.marianhello.bgloc.provider.RawLocationProvider"
    },
    {
        "id": 418,
        "context": 0,
        "level": "INFO",
        "message": "Destroying LocationServiceImpl",
        "timestamp": 1678606203246,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 417,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [2]. startId: [6]",
        "timestamp": 1678606203185,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 416,
        "context": 0,
        "level": "INFO",
        "message": "Stopping self",
        "timestamp": 1678606203166,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 415,
        "context": 0,
        "level": "DEBUG",
        "message": "Task has been removed",
        "timestamp": 1678606203157,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 414,
        "context": 0,
        "level": "INFO",
        "message": "Attempt to stop bg service",
        "timestamp": 1678606203025,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 413,
        "context": 0,
        "level": "INFO",
        "message": "Destroying plugin",
        "timestamp": 1678606203008,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 412,
        "context": 0,
        "level": "INFO",
        "message": "Destroying plugin",
        "timestamp": 1678606202987,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 411,
        "context": 0,
        "level": "INFO",
        "message": "App is no longer visible",
        "timestamp": 1678606202563,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 410,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [4]. startId: [5]",
        "timestamp": 1678606202135,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 409,
        "context": 0,
        "level": "INFO",
        "message": "App will be paused multitasking=true",
        "timestamp": 1678606202101,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 408,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [3]. startId: [4]",
        "timestamp": 1678606014172,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 407,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [8]. startId: [3]",
        "timestamp": 1678606014163,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 406,
        "context": 0,
        "level": "INFO",
        "message": "App will be resumed multitasking=true",
        "timestamp": 1678606014122,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 405,
        "context": 0,
        "level": "INFO",
        "message": "App is visible",
        "timestamp": 1678606014107,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 404,
        "context": 0,
        "level": "INFO",
        "message": "App is no longer visible",
        "timestamp": 1678605992273,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 403,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [STARTED] state. cmdId: [4]. startId: [2]",
        "timestamp": 1678605991541,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 402,
        "context": 0,
        "level": "INFO",
        "message": "App will be paused multitasking=true",
        "timestamp": 1678605991469,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 401,
        "context": 0,
        "level": "DEBUG",
        "message": "Received MSG_ON_SERVICE_STARTED",
        "timestamp": 1678605977026,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 400,
        "context": 0,
        "level": "INFO",
        "message": "Requesting location updates from provider gps",
        "timestamp": 1678605976869,
        "logger": "com.marianhello.bgloc.provider.RawLocationProvider"
    },
    {
        "id": 399,
        "context": 0,
        "level": "INFO",
        "message": "Creating RawLocationProvider",
        "timestamp": 1678605976851,
        "logger": "com.marianhello.bgloc.provider.RawLocationProvider"
    },
    {
        "id": 398,
        "context": 0,
        "level": "DEBUG",
        "message": "Will start service with: Config[distanceFilter=1 stationaryRadius=0.0 desiredAccuracy=0 interval=1000 fastestInterval=1000 activitiesInterval=1000 isDebugging=true stopOnTerminate=true stopOnStillActivity=true startOnBoot=false startForeground=false notificationsEnabled=true locationProvider=2 nTitle=Recording nText=Tracking is active nIconLarge= nIconSmall=app_icon_small nIconColor=#30679a url= syncUrl= syncThreshold=100 httpHeaders={} maxLocations=10000 postTemplate=null]",
        "timestamp": 1678605976832,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 397,
        "context": 0,
        "level": "WARN",
        "message": "Attempt to start unconfigured service. Will use stored or default.",
        "timestamp": 1678605976817,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 396,
        "context": 0,
        "level": "DEBUG",
        "message": "Service in [NOT STARTED] state. cmdId: [0]. startId: [1]",
        "timestamp": 1678605976777,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 395,
        "context": 0,
        "level": "INFO",
        "message": "Network condition changed has connectivity: true",
        "timestamp": 1678605976771,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 394,
        "context": 0,
        "level": "INFO",
        "message": "Creating PostLocationTask",
        "timestamp": 1678605976758,
        "logger": "com.marianhello.bgloc.PostLocationTask"
    },
    {
        "id": 393,
        "context": 0,
        "level": "INFO",
        "message": "Creating LocationServiceImpl",
        "timestamp": 1678605976682,
        "logger": "com.marianhello.bgloc.service.LocationServiceImpl"
    },
    {
        "id": 392,
        "context": 0,
        "level": "INFO",
        "message": "Attempt to start bg service",
        "timestamp": 1678605976653,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 391,
        "context": 0,
        "level": "INFO",
        "message": "User granted requested permissions",
        "timestamp": 1678605976641,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 390,
        "context": 0,
        "level": "DEBUG",
        "message": "Starting service",
        "timestamp": 1678605976633,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 389,
        "context": 0,
        "level": "INFO",
        "message": "Getting current location with timeout:2147483647 maximumAge:9223372036854775807 enableHighAccuracy:false",
        "timestamp": 1678605976582,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 388,
        "context": 0,
        "level": "INFO",
        "message": "App will be resumed multitasking=true",
        "timestamp": 1678605976407,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 387,
        "context": 0,
        "level": "INFO",
        "message": "App will be paused multitasking=true",
        "timestamp": 1678605974075,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 386,
        "context": 0,
        "level": "INFO",
        "message": "Getting current location with timeout:2147483647 maximumAge:9223372036854775807 enableHighAccuracy:false",
        "timestamp": 1678605973869,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 385,
        "context": 0,
        "level": "INFO",
        "message": "Deleting all locations",
        "timestamp": 1678605958640,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 384,
        "context": 0,
        "level": "DEBUG",
        "message": "Registering event listeners",
        "timestamp": 1678605958566,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 383,
        "context": 0,
        "level": "DEBUG",
        "message": "Service configured with: Config[distanceFilter=1 stationaryRadius=0.0 desiredAccuracy=0 interval=1000 fastestInterval=1000 activitiesInterval=1000 isDebugging=true stopOnTerminate=true stopOnStillActivity=true startOnBoot=false startForeground=false notificationsEnabled=true locationProvider=2 nTitle=Recording nText=Tracking is active nIconLarge= nIconSmall=app_icon_small nIconColor=#30679a url= syncUrl= syncThreshold=100 httpHeaders={} maxLocations=10000 postTemplate=null]",
        "timestamp": 1678605958179,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 382,
        "context": 0,
        "level": "INFO",
        "message": "Initializing plugin",
        "timestamp": 1678605958139,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 381,
        "context": 0,
        "level": "INFO",
        "message": "App is no longer visible",
        "timestamp": 1678605897946,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 380,
        "context": 0,
        "level": "INFO",
        "message": "App will be paused multitasking=true",
        "timestamp": 1678605897307,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 379,
        "context": 0,
        "level": "INFO",
        "message": "Deleting all locations",
        "timestamp": 1678605881833,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 378,
        "context": 0,
        "level": "DEBUG",
        "message": "Service configured with: Config[distanceFilter=1 stationaryRadius=0.0 desiredAccuracy=0 interval=1000 fastestInterval=1000 activitiesInterval=1000 isDebugging=true stopOnTerminate=true stopOnStillActivity=true startOnBoot=false startForeground=false notificationsEnabled=true locationProvider=2 nTitle=Recording nText=Tracking is active nIconLarge= nIconSmall=app_icon_small nIconColor=#30679a url= syncUrl= syncThreshold=100 httpHeaders={} maxLocations=10000 postTemplate=null]",
        "timestamp": 1678605881722,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    },
    {
        "id": 377,
        "context": 0,
        "level": "DEBUG",
        "message": "Registering event listeners",
        "timestamp": 1678605880647,
        "logger": "com.tenforwardconsulting.bgloc.cordova.BackgroundGeolocationPlugin"
    },
    {
        "id": 376,
        "context": 0,
        "level": "INFO",
        "message": "Initializing plugin",
        "timestamp": 1678605880559,
        "logger": "com.marianhello.bgloc.BackgroundGeolocationFacade"
    }
]

**Comment by HarelM** [comment]
I don't see in the logs that the app changes the stratForeground state, what makes you say that this is the behavior?

**Comment by veitbjarsch** [comment]
I attached a video. There you can see that every time I minimize the app, the notification will appear. And when I maximize the app it will disappear again. My config looks like the following:
```
{
    locationProvider: BackgroundGeolocation.RAW_PROVIDER,
    desiredAccuracy: BackgroundGeolocation.HIGH_ACCURACY,
    debug: false,
    interval: 1000,
    distanceFilter: 3,
    startForeground: false,
    notificationTitle: null,
    notificationText: null,
    notificationIconColor: '#30679a',
    notificationIconSmall: 'app_icon_small',
}
```

https://user-images.githubusercontent.com/12811012/224883623-26596c43-8e86-4e77-9356-5e8729158053.mp4

**Reply by HarelM** [reply]
Yes this is expected. This is because in order to work in the background, this notification is needed.
If you don't want it to work in the background, use the browser's navigator.

**Reply by veitbjarsch** [reply]
I thought the message is controlled by the value isForeground? If I set it to true it is shown and if I set it to false it is hidden. The geolocation object from the navigator doesn't provide all values that the plugin provides. Also as far I know it is not possible to subscribe to a geolocation change.

**Reply by HarelM** [reply]
No, foreground relates to the priority of the android service.
